doc/rules/basic/numeric_literal_separator.rst

```diff
@@ -0,0 +1,18 @@
+==================================
+Rule ``numeric_literal_separator``
```

Comment:
@keradus to add the separator? Is there any standard encouraging it? I do not see any need of having it.

Reply:
how to add - with `numeric_literal_separator` ;)
PHP migration rulesets are not implementing PSR or anything, only starting using new language syntax/etc with what we consider to be more readable.

Reply:
Ok, I have to be stupid then - because I cannot comprehend how having it is more readable - this would be super confusing for me:
```php
if (\PHP_VERSION_ID >= 80_000) {
    // ...
}
```

Reply:
thus this proposal of mine: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6761#issuecomment-1407665064

one put `70400`? it will be autoconverted to `70_400`. it works as-is, but indeed, not so context-aware here. so, they change it manually to `7_04_00`, and ask to not change numbers with underscores already present

Reply:
Thanks for the use case for leaving them. Seeing this really gives it more value, we should make it an option at least or leave already underscored values as-is.

Reply:
> PHP migration rulesets are not implementing PSR or anything, only starting using new language syntax/etc with what we consider to be more readable.

So far, I see a lot of developer which didn't even know it was possible to write `1_000` and which doesn't find this more readable at all, while I never saw people debating about other PHP new language syntax readability. So I would say adding this in the PHP@7.4 migration could create a lot of negative reaction.

Reply:
I believe we can close this discussion because I agree with @VincentLanglet that there's no point in adding this to the migration rule set. If @keradus decides different, we can always add it later.